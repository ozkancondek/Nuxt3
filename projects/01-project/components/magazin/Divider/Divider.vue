<template>
  <hr :class="classes" :color="color" />
</template>

<script lang="ts">
const COLORS = ["orange", "gray"];
const ORIENTATION = ["horizontal", "vertical"];

export default defineComponent({
  name: "Divider",
  props: {
    color: {
      type: String,
      validator: (value: string) => COLORS.includes(value),
      default: "orange",
    },
    orientation: {
      type: String,
      validator: (value: string) => ORIENTATION.includes(value),
      default: "horizontal",
    },
  },
  setup(props) {
    const classes = computed(() => ({
      "tw-border-0 tw-mx-auto tw-mb-8": true,
      /** color **/
      "tw-bg-orange": props.color === "orange",
      "tw-bg-gray-300": props.color === "gray",
      /** orientation **/
      "tw-w-36 tw-h-0.5": props.orientation === "horizontal",
      "tw-w-0.5 tw-h-36": props.orientation === "vertical",
    }));
    return { classes };
  },
});
</script>
