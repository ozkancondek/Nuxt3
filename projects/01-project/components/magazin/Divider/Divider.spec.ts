import { it, expect } from "vitest";
import { shallowMount } from '@vue/test-utils'
import Divider from '~/components/magazin/Divider/Divider.vue'

it('renders Divider  correctly', () => {
  const wrapper = shallowMount(Divider, {
    mocks: {
      $t: () => {},
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})

// @types/jest
//ts-jest