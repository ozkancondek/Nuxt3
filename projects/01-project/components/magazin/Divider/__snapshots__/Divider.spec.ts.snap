// Vitest Snapshot v1

exports[`renders Divider  correctly 1`] = `
<hr
  class="tw-border-0 tw-mx-auto tw-mb-8 tw-bg-orange tw-w-36 tw-h-0.5"
  color="orange"
/>
`;
